examples/java-matter-controller/BUILD.gn

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+
+import("${build_root}/config/android_abi.gni")
+import("${chip_root}/build/chip/java/rules.gni")
+import("${chip_root}/build/chip/tools.gni")
+
+android_binary("java-matter-controller") {
+  output_name = "java-matter-controller"
+
+  deps = [
+    ":android",
```

Comment:
Is this supposed to be permanent? I imagine java-controller would be runnable on linux, in which case it should use the linux platform not android subsystem.

Should we add some TODO here?

Reply:
We plan to migrate to JDK from Android SDK after https://github.com/project-chip/connectedhomeip/issues/23133 is addressed

Reply:
Since we can link CHIPController.jar and run the Java application built with android binary x84 directly on Linux, we plan to  continue this route at this moment until we hit the problem